src/core/effects/qgspainteffect.cpp

```diff
@@ -322,3 +322,41 @@ void QgsDrawSourceEffect::readProperties( const QgsStringMap &props )
   mEnabled = props.value( QStringLiteral( "enabled" ), QStringLiteral( "1" ) ).toInt();
   mDrawMode = static_cast< QgsPaintEffect::DrawMode >( props.value( QStringLiteral( "draw_mode" ), QStringLiteral( "2" ) ).toInt() );
 }
+
+
+//
+// QgsEffectPainter
+//
+
+QgsEffectPainter::QgsEffectPainter( QgsRenderContext& renderContext )
+    : mRenderContext( renderContext ),
+    mEffect( nullptr )
+{
+  mPainter = renderContext.painter();
+  mPainter->save();
+}
+
+QgsEffectPainter::QgsEffectPainter( QgsRenderContext& renderContext, QgsPaintEffect* effect )
+    : mRenderContext( renderContext ),
```

Comment:
one more thing, for consistency can you move the comma to the next line?

    : mRenderContext( ... )
    , mEffect( ... )